### PR TITLE
Improve IntDivThrowTypeExtension to support IntRange

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Type/Php/IntdivThrowTypeExtension.php [hunk 1]
```diff
@@ -29,39 +28,19 @@ public function getThrowTypeFromFunctionCall(FunctionReflection $functionReflect
...
-			}
-
-			$valueType = TypeCombinator::remove($valueType, $constantScalarType);
-		}
-
-		if (!$valueType instanceof NeverType) {
-			$containsMin = true;
-		}
+		$valueType = $scope->getType($funcCall->getArgs()[0]->value)->toInteger();
+		$containsMin = $valueType->acceptsWithReason(new ConstantIntegerType(PHP_INT_MIN), true);
```
Comment (ondrejmirtes) [inline src/Type/Php/IntdivThrowTypeExtension.php]: isSuperTypeOf() should be used instead of accepts()

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/IntdivThrowTypeExtension.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Type/Php/IntdivThrowTypeExtension.php]: Updated.

Out of curiosity, what's the diff. Seems like both works.

Comment (ondrejmirtes) [reply src/Type/Php/IntdivThrowTypeExtension.php]: https://github.com/phpstan/phpstan/discussions/11460

Comment (VincentLanglet) [reply src/Type/Php/IntdivThrowTypeExtension.php]: Thanks

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**